### PR TITLE
Tests for TF2.0

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Fix tests
- Tests for random seed are ignored right now, will do another PR for them as it is a bigger change
- Rename `EmbeddingPolicy` to `TEDPolicy`
- Rename `EmbeddingIntentClassifier` to `DIETClassifier`

part of https://github.com/RasaHQ/research/issues/62

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/classifiers/DIET_classifier.py [hunk 1]
```diff
@@ -570,7 +573,6 @@ def train(
             self.component_config[EVAL_NUM_EXAMPLES],
             self.component_config[EVAL_NUM_EPOCHS],
             batch_strategy=self.component_config[BATCH_STRATEGY],
```
Comment (Ghostvv) [inline rasa/nlu/classifiers/DIET_classifier.py]:
```suggestion
            self.component_config[BATCH_STRATEGY],
```
I think it is intuitive like this

File: rasa/utils/tensorflow/tf_models.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ class RasaModel(tf.keras.models.Model):
     Cannot be used as tf.keras.Model
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, random_seed: Optional[int], *args, **kwargs):
```
Comment (Ghostvv) [inline rasa/utils/tensorflow/tf_models.py]: I don't think we need `*args` here. Let's set `random_seed` to `None` by default

File: tests/core/test_policies.py [hunk 3]
```diff
@@ -207,14 +214,14 @@ def create_policy(self, featurizer, priority):
...
 
-    # TODO test tf config
-    # def test_tf_config(self, trained_policy, tmpdir):
-    #     # noinspection PyProtectedMember
-    #     assert trained_policy.session._config == session_config()
-    #     trained_policy.persist(tmpdir.strpath)
-    #     loaded = trained_policy.__class__.load(tmpdir.strpath)
-    #     # noinspection PyProtectedMember
-    #     assert loaded.session._config == session_config()
+    @pytest.mark.skip(reason="We need to fix tf.config!")
```
Comment (Ghostvv) [inline tests/core/test_policies.py]: let's keep `TODO` comment as well

File: docs/nlu/components.rst [hunk 4]
```diff
@@ -147,7 +147,7 @@ ConveRTFeaturizer
     `ConveRT <https://github.com/PolyAI-LDN/polyai-models>`_ model.
 :Outputs:
     nothing, used as an input to intent classifiers and response selectors that need intent features and response
-    features respectively (e.g. ``EmbeddingIntentClassifier`` and ``ResponseSelector``)
+    features respectively (e.g. ``DIETClassifier`` and ``ResponseSelector``)
```
[the remark this answers is not in the document]
Comment (Ghostvv) [reply docs/nlu/components.rst]: let's leave response selector staff to a separate PR

[the remark this answers is not in the document]
Comment (tabergma) [reply docs/nlu/components.rst]: 👍

File: rasa/nlu/classifiers/DIET_classifier.py [hunk 5]
```diff
@@ -438,11 +440,16 @@ def _create_model_data(
...
-            _sparse, _dense = self._extract_and_add_features(e, TEXT_ATTRIBUTE)
-            if _sparse is not None:
-                X_sparse.append(_sparse)
-            if _dense is not None:
-                X_dense.append(_dense)
+            if (
+                label_attribute is None
+                or label_attribute == INTENT_ATTRIBUTE
+                or e.get(label_attribute)
+            ):
```
[the remark this answers is not in the document]
Comment (tabergma) [reply rasa/nlu/classifiers/DIET_classifier.py]: For response selection, if we train the response selector, but not all examples have the response attribute set, we collect more text features than label features. That will raise an error later when we add the data to the model data as the dimensions differ.

[the remark this answers is not in the document]
Comment (Ghostvv) [reply rasa/nlu/classifiers/DIET_classifier.py]: then, I think we don't need `or label_attribute == INTENT_ATTRIBUTE` ?

[the remark this answers is not in the document]
Comment (Ghostvv) [reply rasa/nlu/classifiers/DIET_classifier.py]: because if for some reason intent examples don't have `INTENT_ATTRIBUTE` it will also raise an error

[the remark this answers is not in the document]
Comment (tabergma) [reply rasa/nlu/classifiers/DIET_classifier.py]: we only don't need it if every training example has an intent attribute, but I guess we can assume that

[the remark this answers is not in the document]
Comment (Ghostvv) [reply rasa/nlu/classifiers/DIET_classifier.py]: Then we have a problem 🙈, I think theoretically we allow examples without intent, but then entities will not be trained

[the remark this answers is not in the document]
Comment (tabergma) [reply rasa/nlu/classifiers/DIET_classifier.py]: Yeah, we can only train intent and entities together, if all examples have an intent set, so I guess we can remove it. The check for `None` is needed during prediction. If we just want to train entities, we can set the attribute to `None`. That should work.

[the remark this answers is not in the document]
Comment (Ghostvv) [reply rasa/nlu/classifiers/DIET_classifier.py]: then if `intent_classification` is `None`, we should set `label_attribute` to `None`
